src/components/Router.js

```diff
@@ -58,10 +58,11 @@ class Router extends PureComponent {
         this.route = {
           ...result.route,
           ...this.props.router.route,
+          ...(status ? { status } : {}),
         }
         if (this.props.onChange) {
           this.props.onChange({
-            route: result.route,
+            route: this.route,
```

Comment:
You sure this needs to be changed? I'm not sure what the side effects of changing this to include all route data is.

Reply:
Yeah, this is a bit confusing. I was thinking that `this.route` should have been passed to `onChange`, because we merge `this.props.router.route` into it in the lines above. I'm not sure what that is doing, either. But it seems we should have the latest updated data in both places.

I was not really sure if status should be updated here in the first place. Maybe it makes more sense to send it to `transition.js` and then have it update the `route.status` there?

Btw, the only other place I see `onChange` used is in AG.js client render (`src/lib/client/render.js`)